xdsl/dialects/riscv_ssa.py

```diff
@@ -0,0 +1,711 @@
+from __future__ import annotations
```

Comment:
I'd recommend putting this file in experimental for the time being.

Reply:
Mostly because it's a very interesting case of a lower-level semantic. My feeling is that it would benefit from being reimplemented at a higher level of abstraction, without any mentions of registers. It would probably also be beneficial to keep the implementation in sync with the translation to the emulation form. Maybe we'd be better off putting it into a different module altogether, maybe together with the toy dialect in the docs, and with the rest of the compiler. That way we have at least one true end-to-end compiler in the xDSL repo that would be helpful for a bunch of different reasons.

Reply:
A side comment is that this dialect and the rest of the work could definitely wait until the end of the paper deadlines that are higher priority for the team for the near future. It would be good to give this work the attention it deserves, and not compete with the other parallel work

Reply:
Sure, that's a good way to also do it. I didn't think that it would be this heavily discussed, but yeah. The more I think about it the more I agree that this is not something we want to have in main in it's current form!